doc/source/getting_started/develop_pymapdl.rst

```diff
@@ -0,0 +1,337 @@
+
+.. _developing_pymapdl:
+
+===============
+Develop PyMAPDL
+===============
+
+You can help PyMAPDL library by fixing an bug or developing a new feature.
+To do that, you need to setup the repository in your local machine as it is explained
+in the following sections.
+
+
+Cloning the PyMAPDL repository
+==============================
+
+To clone PyMAPDL repository, you'll need to install a version control system (VCS) such as Git.
+Later run this code to clone the latest development version of PyMAPDL:
+
+.. code:: console
+
+    git clone https://github.com/ansys/pymapdl
+    cd pymapdl
+
+
+
+Create a Python virtual environment
+===================================
+
+It is recommended you install PyMAPDL in its own virtual environment to avoid
+dependency conflicts and to easier manage upgrades.
+Visit `Setting up your development environment <dev_guide_setup_your_environment_>`_
+for detailed information on how to install Python and create a virtual environment.
+
+Install PyMAPDL in development mode
+===================================
+
+Install the latest version of PyMAPDL in development mode:
+
+
+.. code:: console
+
+    cd pymapdl
+    pip install pip -U
+    pip install -e .
+
+
+If you are going to do testing, you need to install the testing dependencies:
+
+
+.. code:: console
+
+    pip install -e '.[tests]'
+
+
+Develop PyMAPDL
+===============
+
+.. epigraph:: *Now it is time to develop PyMAPDL!*
+
+Developing code in a repository, particularly when using version control systems
+like Git, involves a set of essential guidelines to ensure efficient
+collaboration, code management, and tracking changes. Here are the main
+guidelines for developing code in a repository:
+
+#. **Use Branches**: Create branches for different features, bug fixes, or
+   experiments. This keeps changes isolated and facilitates parallel
+   development.
+
+#. **Write Descriptive Commit Messages**: Provide clear and concise commit
+   messages that explain the purpose and context of the changes. Follow a
+   consistent style.
+
+#. **Commit Frequently**: Make small, meaningful commits frequently. Avoid
+   making a large number of unrelated changes in a single commit.
+
+#. **Pull Before You Push**: Always update your local branch with the latest
+   changes from the remote repository before pushing your own changes to avoid
+   conflicts.
+
+#. **Use Pull Requests (PRs)**: Use PRs to submit your changes for review.
+   This allows for discussion and validation before merging into the main branch.
+
+#. **Write good documentation**: Maintain clear and up-to-date documentation for your
+   contribution or changes, including README files, comments in code, and relevant project
+   documentation. Make sure you update the related documentation as well.
+   Follow `numpydoc <numpydoc_>`_ convention for documenting code.
```

Comment:
```suggestion
   Follow the `numpydoc <numpydoc_>`_ convention for documenting code.
```

Reply:
Good call.

Reply:
There might be more readmes in different directories. It is true that in our project, there are few (the main one, and one in docker directory and another one in `.devcontainer` directory)